### PR TITLE
Commented codes related to mongroup functionality.

### PR DESCRIPTION
**Pull request by truewis**
#### Summary
Features "Added comments to mongroup.h and related functions."

#### Purpose of change

Improving the code quality of the existing monster hoard functionality is important for the future implementation of the feature. 

#### Describe the solution

Added comments.

#### Testing

No test performed:
Did not write anything besides comments, which could be verified easily.

### CODE REVIEW

File: src/mongroup.h [hunk 1]
```diff
@@ -86,26 +86,39 @@ struct MonsterGroupResult {
...
+
+    /*replaces this group after a period of time when exploring an unexplored portion of the map. Used for monster evolution. */
     bool replace_monster_group = false;
+    /* The monster group to replace this if replace_monster_group == true. */
     mongroup_id new_monster_group;
+    /* Time until replacement if replace_monster_group == true. */
     time_duration monster_group_time = 0_turns;
-    bool is_safe = false; /// Used for @ref mongroup::is_safe()
+
+    /** Encapsulated by mongroup::is_safe() , but currently not used anywhere.*/
```
Comment (Night-Pryanik) [inline src/mongroup.h:106]:
```suggestion
    /** Encapsulated by mongroup::is_safe(), but currently not used anywhere.*/
```

File: src/mongroup.h [hunk 2]
```diff
@@ -86,26 +86,39 @@ struct MonsterGroupResult {
...
     std::map<holiday, int> event_freq; // total freq for each event
     // Get the total frequency of entries that are valid for the specified event.
     // This includes entries that have an event of "none". By default, use the current holiday.
     int event_adjusted_freq_total( holiday event = holiday::num_holiday ) const;
 };
 
+/**
+ * A mongroup is a group of monsters that are spawned together. There are two uses for mongroups:
+ * 1. spawning new monsters.
+ * 2. unload monsters out of sight of the player into hoards and spawn them back in when the player is close enough.
```
Comment (Night-Pryanik) [inline src/mongroup.h:118]:
```suggestion
 * 2. unload monsters out of sight of the player into hordes and spawn them back in when the player is close enough.
```

File: src/mongroup.h [hunk 3]
```diff
@@ -86,26 +86,39 @@ struct MonsterGroupResult {
...
+ */
 struct mongroup {
+    /** The type of the mongroup, which is only used to spawn new monsters and not for hoards. */
```
Comment (Night-Pryanik) [inline src/mongroup.h:121]:
```suggestion
    /** The type of the mongroup, which is only used to spawn new monsters and not for hordes. */
```

File: src/mongroup.h [hunk 4]
```diff
@@ -115,12 +128,25 @@ struct mongroup {
...
+    /** location the horde is interested in. Do not set this directly, use set_target instead. */
+    point_abs_sm target;
+    /** abs target for nemesis hordes. */
+    point_abs_sm nemesis_target; //
+    /** interest to target in percents. Used to determine movement speed(overmap::move_hordes), or reaction to signal sources(overmap::signal_hordes), etc.*/
+    int interest = 0;
+
+    /** If true, overmap::process_mongroups decreases the population by a fixed ratio every time the function is called. */
     bool dying = false;
+    /** If false, this group is not a hoard, and instead is likely an attempt to spawn new monsters through map::spawn_monsters_submap from overmap_buffer. */
```
Comment (Night-Pryanik) [inline src/mongroup.h:149]:
```suggestion
    /** If false, this group is not a horde, and instead is likely an attempt to spawn new monsters through map::spawn_monsters_submap from overmap_buffer. */
```

Comment (truewis) [reply src/mongroup.h:149]: Thank you for the correction!

**Pull request merged**